### PR TITLE
Add `mc admin group` command

### PR DESCRIPTION
**Pull request by donatello**
This change is in preparation to add user-groups support to MinIO
server.

- Adds add, remove, enable/disable, list and info subcommands for the
  group command

- Removes the ability to set the user-policy in a user add command

### CODE REVIEW

**Comment by donatello** [comment]
To locally test this PR, checkout this code and update the replace directive in go.mod https://github.com/minio/mc/pull/2841/files#diff-37aff102a57d3d7b797f152915a6dc16R49 to match to the local checkout of PR https://github.com/minio/minio/pull/7981 of the MinIO server.

**Comment by donatello** [comment]
**NOTE** This PR will not build on CI until minio/minio#7981 is merged.

**Comment by kannappanr** [comment]
@donatello minio/minio#7981 has been merged. Can you fix `go.mod` ?

**Comment by donatello** [comment]
> @donatello minio/minio#7981 has been merged. Can you fix go.mod ?

Done!

File: cmd/admin-group-list.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// checkAdminGroupListSyntax - validate all the passed arguments
+func checkAdminGroupListSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "list", 1) // last argument is exit code
+	}
+}
+
+// mainAdminGroupList is the handle for "mc admin remove remove" command.
```
Comment (krisis) [inline cmd/admin-group-list.go]:
```suggestion
// mainAdminGroupList is the handle for "mc admin group list" command.
```

File: cmd/admin-group-info.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+// checkAdminGroupInfoSyntax - validate all the passed arguments
+func checkAdminGroupInfoSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 2 {
+		cli.ShowCommandHelpAndExit(ctx, "info", 1) // last argument is exit code
+	}
+}
+
+// mainAdminGroupInfo is the handle for "mc admin remove remove" command.
```
Comment (krisis) [inline cmd/admin-group-info.go]:
```suggestion
// mainAdminGroupInfo is the handle for "mc admin group info" command.
```

File: cmd/admin-group-enable.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+// checkAdminGroupEnableSyntax - validate all the passed arguments
+func checkAdminGroupEnableSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 2 {
+		cli.ShowCommandHelpAndExit(ctx, "enable", 1) // last argument is exit code
+	}
+}
+
+// mainAdminGroupEnableDisable is the handle for "mc admin remove remove" command.
```
Comment (krisis) [inline cmd/admin-group-enable.go]: shouldn't this be "mc admin group enable|disable" command?

File: cmd/admin-group-remove.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// checkAdminGroupRemoveSyntax - validate all the passed arguments
+func checkAdminGroupRemoveSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) < 2 {
+		cli.ShowCommandHelpAndExit(ctx, "remove", 1) // last argument is exit code
+	}
+}
+
+// mainAdminGroupRemove is the handle for "mc admin remove remove" command.
```
Comment (krisis) [inline cmd/admin-group-remove.go]:
```suggestion
// mainAdminGroupRemove is the handle for "mc admin group remove" command.
```

File: cmd/admin-group-remove.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		members = append(members, args.Get(i))
+	}
+	gAddRemove := madmin.GroupAddRemove{
+		Group:    args.Get(1),
+		Members:  members,
+		IsRemove: true,
+	}
+
+	e := client.UpdateGroupMembers(gAddRemove)
+	fatalIf(probe.NewError(e).Trace(args...), "Cannot remove new group")
```
Comment (krisis) [inline cmd/admin-group-remove.go]: Why does the error message say "new group"?

File: cmd/admin-group-enable.go (hunk 3 above)
Comment (donatello) [reply cmd/admin-group-enable.go]: Yep, done.

File: cmd/admin-group-remove.go (hunk 5 above)
Comment (donatello) [reply cmd/admin-group-remove.go]: Fixed.

**Pull request merged**